### ISSUE TITLE
TypeError: can only concatenate list (not "filter") to list

### ISSUE THREAD

**Issue by monajalal** [HazyResearch/cs145-notebooks-2016#4]
For cs145-notebooks-2016/lecture-16/Activity-16-1.ipynb
I get the following error in Python 3.6.1:
<img width="898" alt="screen shot 2017-09-19 at 2 49 24 pm" src="https://user-images.githubusercontent.com/1892917/30609600-d735e6b4-9d49-11e7-8fe4-ee1a99564d17.png">
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-18-4e1a4ac4ce1b> in <module>()
----> 1 j = NJoin(R, S)
      2 render_markdown(j)
      3 print(get_result(j))

/Users/mona/CS460_660/relation_algebra.py in __init__(self, op1, op2)
    164         self.common = s1.intersection(s2)
    165         self.op_str = "$\Join_{{{0}}}$".format(','.join(self.common))
--> 166         OpBase.__init__(self, op1.schema + filter(lambda x : x not in self.common, op2.schema), [op1,op2])
    167         self.count_reads = True
    168 

TypeError: can only concatenate list (not "filter") to list
```
All the other codes before this ran perfectly. Any idea what needs to be fixed?

**Comment by ajratner** [comment]
I think this is a Python 3 issue- the notebooks were written for Python 2.  If there's any interest in modifying to support Python 3, we'd be happy to review a PR!

**Comment by monajalal** [comment]
@ajratner  I fixed it for Anaconda Python 3.6.2. You can find it here:
https://github.com/monajalal/CS460_660/tree/master/lab3

**Comment by ajratner** [comment]
Great! If you want to submit as a PR would be happy to review, either way
glad you got by this issue!
